module/VuFind/src/VuFind/I18n/SorterTrait.php

```
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Trait SorterTrait
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Moravian Library 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  VuFind\Service
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\I18n;
+
+/**
+ * Trait SorterTrait
+ *
+ * @category VuFind
+ * @package  VuFind\Service
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+trait SorterTrait
+{
+    /**
+     * Sorter
+     *
+     * @var ?Sorter
+     */
+    protected $sorter = null;
+
+    /**
+     * Set the sorter
+     *
+     * @param Sorter $sorter Sorter service
+     *
+     * @return void
+     */
+    public function setSorter(Sorter $sorter): void
+    {
+        $this->sorter = $sorter;
+    }
+
+    /**
+     * Get the sorter
+     *
+     * @return Sorter
+     */
+    public function getSorter(): Sorter
+    {
+        return $this->sorter ?? new Sorter();
```

Might it be better to throw an exception or log a warning if there is no sorter defined? I'm concerned about a scenario where forgetting to inject a dependency causes inconsistent sorting behavior which leads to confusion. Of course, maybe allowing this to work seamlessly is a more important priority... I'm certainly open to discussion on this!

---

I think it is not so important. It throws an exception now.